src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/Source.cs

```diff
@@ -206,41 +211,24 @@ private static RoslynTrigger GetRoslynTrigger(AsyncCompletionData.CompletionTrig
             return item;
```

Comment:
```
            var item = new VSCompletionItem(
                displayText: roslynItem.DisplayText,
                 source: this,
                 icon: new ImageElement(new ImageId(imageId.Guid, imageId.Id), roslynItem.DisplayText),
                 filters: filters,
                 suffix: string.Empty,
                 insertText: insertionText,
```

This API doesn't make sense.  insertiontext isn't something that can be known at VSCompletionItem creation time.  It's dynamically computed when the item is inserted, depending on what the service/provider decides is best.

Reply:
@AmadeusW, could you please take a look?

Reply:
I will correct our commit algorithm to take the insertionText dynamically running GetChangesAsync in all cases. However, if the editor wants us to provide a insertionText now, this could be our best approach. @amwieczw

---
In reply to: [222519089](https://github.com/dotnet/roslyn/pull/29016#discussion_r222519089) [](ancestors = 222519089)